PreFreeSurfer/scripts/AnatomicalAverage.sh

```diff
@@ -4,126 +4,170 @@
 #  Usage Description Function
 # ------------------------------------------------------------------------------
 
-script_name=$(basename "${0}")
+# script_name=$(basename "${0}")
 
-Usage() {
-	cat <<EOF
+set -eu
 
-Usage: ${script_name} [options] <image1> ... <imageN>
+pipedirguessed=0
+if [[ "${HCPPIPEDIR:-}" == "" ]]
+then
+    pipedirguessed=1
+    #fix this if the script is more than one level below HCPPIPEDIR
+    export HCPPIPEDIR="$(dirname -- "$0")/.."
+fi
+
+source "${HCPPIPEDIR}/global/scripts/debug.shlib" "$@"         # Debugging functions; also sources log.shlib
+source "$HCPPIPEDIR/global/scripts/newopts.shlib" "$@"
+
+opts_SetScriptDescription "Tool for creating a Anatomical Average"
+
+opts_AddMandatory '--output' 'output' 'name' "output basename"
+
+opts_AddMandatory '--image-list' 'imagesStr' 'image1@image2...' "list of images to average"
+
+#Optional Args
+opts_AddOptional '--standard-image' 'StandardImage' 'image' "standard image (e.g. MNI152_T1_2mm)" "$FSLDIR/data/standard/MNI152_T1_2mm.nii.gz"
+
+opts_AddOptional '--standard-mask' 'StandardMask' 'image' "standard brain mask (e.g. MNI152_T1_2mm_brain_mask_dil)" "$FSLDIR/data/standard/MNI152_T1_2mm_brain_mask_dil.nii.gz"
+
+opts_AddOptional '--brain-size' 'BrainSizeOpt' 'number' "pass a brain size value to robustfov"
 
-Compulsory arguments
-  -o <name>        : output basename
-Optional arguments
-  -s <image>       : standard image (e.g. MNI152_T1_2mm)
-  -m <image>       : standard brain mask (e.g. MNI152_T1_2mm_brain_mask_dil)
-  -n               : do not crop images
-  -w <dir>         : local, temporary working directory (to be cleaned up - i.e. deleted)
-  --noclean        : do not run the cleanup
-  -v               : verbose output
-  -h               : display this help message
+opts_AddOptional '--crop' 'crop' 'yes OR no' "whether to crop images, default yes" "yes"
 
-e.g.:  ${script_name} -n -o output_name im1 im2
+opts_AddOptional '--working-dir' 'wdir' 'dir' "temporary working directory" 
 
-Note that N>=2 (i.e., there must be at least two images in the list)
+opts_AddOptional '--cleanup' 'cleanup' 'yes OR no' "whether to delete the working directory (if set via --working-dir), default yes" "yes"
 
-EOF
-}
+#TSC: verbose is redundant given opts_ShowValues, remove it
+#opts_AddOptional '--verbose' 'verbose' 'yes OR no' "verbose output" "no"
```

Comment:
Please delete the commented out old code, or things that aren't needed, like this.

Reply:
Done